### ISSUE TITLE
Add build instructions

### ISSUE THREAD

**Issue by IBPX** [Siile/Ninslash#15]
There should be instructions on how to build Ninslash in readme.txt.

**Comment by Henningstone** [comment]
I don't think this is really necessary since it doesn't differ on how to build teeworlds at all, does it? :o

**Comment by IBPX** [comment]
Well, Teeworlds doesn't have build instructions in the readme either. You have to go to their website.

I think it would be nice to have at least a few lines telling you what to do to build it in the readme.

**Comment by Siile** [comment]
I only know build instructions for windows, and that involves visual studio, python, bam and some black magic. I don't mind keeping the secrets within teeworlds community for a while, but I have nothing against linking or writing instructions either.

**Comment by pelya** [comment]
I dislike bam, it does absolutely nothing that you cannot achieve via CMake or a primitive makefile, and there's no way to change compiler command or add options like -flto or -farch=core2 without editing main bam file. Bam is used only to compile Teeworlds, no other project uses it, and it has very little documentation.

**Comment by Henningstone** [comment]
Bam simplifies multiplatform compilation quite a bit, and since the makefiles are plain lua you can customize the build process even more... Also (just out of curiosity), what are you missing in the [docs](http://matricks.github.io/bam/bam.html) ?

**Comment by pelya** [comment]
So is CMake. But I don't want to edit versioned files just to add few
options for my compiler, because git pull stops working until I create a
commit from that change, I should be able to do that using environment
variables, as all other build systems do.

Docs are missing literally that, try googling 'cmake cross compile' and
you'll get a hundred links, try googling 'bam cross compile' and the only
result you'll get is 'BAM there goes 30 minutes of my time and I haven't
done anything productive', and it's not even about 'bam the build tool'.
